apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.sql

```diff
@@ -6,6 +6,9 @@ SELECT
       ap.name
       ORDER BY
         ap.name
+    ) FILTER (
+      WHERE
+        ap.id IS NOT NULL
```

Comment:
I believe the enclosing `coalesce` might actually be unnecessary now?

Reply:
You still need the `coalesce` (I tested it).

I do wonder whether it might not be clearer to write this with a sub-select as:
```
select
    to_jsonb(i.*) AS institution,
    coalesce((
        select
            jsonb_agg(ap.name order by ap.name)
        from
            institution_authn_providers as iap
            join authn_providers as ap on (ap.id = iap.authn_provider_id)
        where iap.institution_id = i.id
    ), '[]'::jsonb) as authn_providers
from institutions as i
order by i.short_name;
```

Reply:
Also do we need an ORDER BY on the institutions?

Reply:
Thanks for testing the coalesce. I do think your version is marginally cleaner, I switched to that in 97c2d55.

As for ordering, I think it's been ordering by `institutions.id` by default. Frankly, I've grown accustomed to this, and I kind of like it since it puts the "special" institutions (Default and LTI) at the top. I added an explicit `ORDER BY` clause in my latest commit too.

Reply:
I think that without an ORDER BY you'll often get an `id` order by default because it scans the table in the order rows were originally inserted. But this is not guaranteed -- for example, if the table gets big enough that PG starts doing parallel scans using multiple workers, it won't re-order the combined results by `id` anymore.

Reply:
I can only hope that we eventually have enough institutions that Postgres has to use multiple workers to scan them 😉